packages/amplify-category-function/src/provider-utils/awscloudformation/utils/convertLayersTypes.ts

```diff
@@ -0,0 +1,52 @@
+import { ExternalLayer, LambdaLayer, ProjectLayer } from 'amplify-function-plugin-interface';
+import { convertProjectLayer } from './layerArnConverter';
+
+const LAYER_ARN_KEY = 'Fn::Sub';
+
+export const convertProjectLayersToExternalLayers = (lambdaLayers: LambdaLayer[], envName: string): LambdaLayer[] => {
+  const modifiedLambdaLayers: LambdaLayer[] = [];
+  lambdaLayers.forEach(layer => {
+    if (layer.type === 'ProjectLayer' && layer.env !== envName && !layer.isLatestVersionSelected) {
+      const convertLayer: ExternalLayer = {
+        type: 'ExternalLayer',
+        arn: convertProjectLayer(layer, envName),
+      };
+      modifiedLambdaLayers.push(convertLayer);
+    } else {
+      const convertLayer: ProjectLayer = {
+        type: 'ProjectLayer',
+        resourceName: (layer as ProjectLayer).resourceName,
```

Comment:
cast it once and just used the casted instance as this becomes verbose even for 2 usages.

Reply:
done